core/src/main/kotlin/org/evomaster/core/search/gene/SqlPrimaryKeyGene.kt

```diff
@@ -55,8 +55,9 @@ class SqlPrimaryKeyGene(name: String,
 
     override fun getVariableName() = gene.getVariableName()
 
-    override fun flatView(excludePredicate: (Gene)-> Boolean): List<Gene> {
-        return if(excludePredicate(this)) listOf(this) else listOf(this).plus(gene.flatView(excludePredicate))
+    override fun flatView(excludePredicate: (Gene) -> Boolean): List<Gene> {
+        return if (excludePredicate(this)) listOf() else
+            listOf(this).plus(gene.flatView(excludePredicate))
```

Comment:
isn't this a regression bug where ` listOf(this)` is wrongly replaced with ` listOf()`?

Reply:
You're right, sorry for this.